pkg/gui/controllers.go

```diff
@@ -41,6 +42,7 @@ func (gui *Gui) resetControllers() {
 			func() *cherrypicking.CherryPicking { return gui.State.Modes.CherryPicking },
 			rebaseHelper,
 		),
+		Upstream: helpers.NewUpstreamHelper(helperCommon, model, gui.getSuggestedRemote, suggestionsHelper.GetRemoteBranchesSuggestionsFunc),
```

Comment:
I'd take this one step further and define getSuggestedRemote within the new helper file, given that it relates to the upstream. That way we can delete `misc.go` (and our gui struct will be slimmer)